### PR TITLE
The start of new callback design

### PR DESCRIPTION
**Pull request by tilen1976**
First part of new variables callback design:
      - Open dataset:
         -   Based on variables in dataset a layout is build with one accordion per variabel
         -   Editable fields are based on component type i display_new_variables.py
         -   Editable fields are wrapped in Form element
         -   value is set to empty string to avoid "undefined" issue (element is changed from controlled to uncontrolled)
 
Missing:
     - Get values for dropdowns
     - Save values

### CODE REVIEW

File: src/datadoc/frontend/components/resources_test_new_variables.py [hunk 1]
```diff
@@ -2,120 +2,82 @@
...
-                className="variables-input-group",
-            ),
-            build_input_dropdowns("Datatype", data_type),
+            (
+                i.component(
+                    label=i.display_name,
+                    disabled=not i.editable,
+                    className="variabels-input",
+                    id={
+                        "type": NEW_DATASET_METADATA_INPUT,
```
Comment (mmwinther) [inline src/datadoc/frontend/components/resources_test_new_variables.py:32]: Should we not also use the name of the variable in the `id` dict? I think that's what has caused the problems with duplicate IDs.

File: src/datadoc/frontend/callbacks/register_callbacks.py [hunk 2]
```diff
@@ -178,3 +180,25 @@ def register_new_variables_tab_callbacks(app: Dash) -> None:
...
+        respons_list: list = []
+        index = 0
+        if n_clicks and n_clicks > 0:
+            for v in state.metadata.meta.variables:
+                index += 1
+                response = get_display_values(v, state.current_metadata_language)
+                respons_list.append(
+                    build_ssb_accordion(
+                        response["short_name"],
+                        {"type": "variabels-accordion", "id": index},
```
Comment (mmwinther) [inline src/datadoc/frontend/callbacks/register_callbacks.py:201]: Typo, needs to be updated a few other places too

```suggestion
                        {"type": "variables-accordion", "id": index},
```

**Review by mmwinther — CHANGES_REQUESTED** [review]
Looks good and works well, a couple of comments

File: src/datadoc/frontend/components/resources_test_new_variables.py [hunk 3]
```diff
@@ -2,120 +2,82 @@
...
+                        "id": i.identifier,
+                    },
+                    value="",
+                    type=i.presentation,
+                )
+                if i.component == ssb.Input
+                else i.component(
+                    className="variabels-input",
+                    id={
+                        "type": NEW_DATASET_METADATA_INPUT,
```
Comment (mmwinther) [inline src/datadoc/frontend/components/resources_test_new_variables.py:42]:
```suggestion
                        "type": VARIABLES_METADATA_INPUT,
```

File: src/datadoc/frontend/components/resources_test_new_variables.py [hunk 4]
```diff
@@ -2,120 +2,82 @@
...
-# dropdown: datatype,variabelens, DPI,temporalitetstype,
-variables_details_test_obligatory = [
-    "Kortnavn",
-    "Navn",
-    "Datatype",
-    "Variabelens rolle",
-    "Definition URI",
-    "DPI",
-]
+NEW_DATASET_METADATA_INPUT = "new-dataset-metadata-input"
```
Comment (mmwinther) [inline src/datadoc/frontend/components/resources_test_new_variables.py:17]:
```suggestion
NEW_DATASET_METADATA_INPUT = "variables-metadata-input"
```

**Review by mmwinther — COMMENTED** [review]
A couple of other small things 😇

File: src/datadoc/frontend/components/resources_test_new_variables.py (hunk 1 above)
Comment (tilen1976) [reply src/datadoc/frontend/components/resources_test_new_variables.py]: Here I already use name as part of the ID (i.identifier)

**Review by mmwinther — APPROVED** [review]
Nice work!

**Pull request merged**